app/namespace-watcher.go

```diff
@@ -45,77 +70,149 @@ func createLimitRange(clientset *kubernetes.Clientset, namespaceName string) err
 				{
 					Type: corev1.LimitTypePod,
 					Max: corev1.ResourceList{
-						corev1.ResourceCPU:              resource.MustParse(cpuLimitMax),
-						corev1.ResourceMemory:           resource.MustParse(memoryLimitMax),
-						corev1.ResourceEphemeralStorage: resource.MustParse(ephemeralStorageLimitMax),
+						corev1.ResourceCPU:              cpuLimitMax,
+						corev1.ResourceMemory:           memLimtMax,
+						corev1.ResourceEphemeralStorage: ephemeralStorageMax,
 					},
 					Min: corev1.ResourceList{
-						corev1.ResourceCPU:              resource.MustParse(cpuLimitMin),
-						corev1.ResourceMemory:           resource.MustParse(memoryLimitMin),
-						corev1.ResourceEphemeralStorage: resource.MustParse(ephemeralStorageLimitMin),
+						corev1.ResourceCPU:              cpuLimitMin,
+						corev1.ResourceMemory:           memLimtMin,
+						corev1.ResourceEphemeralStorage: ephemeralStorageMin,
 					},
 				},
 			},
 		},
 	}
 
-	_, err := clientset.CoreV1().LimitRanges(namespaceName).Create(context.Background(), limitRange, metav1.CreateOptions{})
+	// Get the existing LimitRange
+	existingLimitRange, err := clientset.CoreV1().LimitRanges(namespaceName).Get(context.Background(), limitRange.ObjectMeta.Name, metav1.GetOptions{})
 	if err != nil {
-		return err
+		if errors.IsNotFound(err) {
+
+			//Create LimitRange if it doesn't exist
+			_, err = clientset.CoreV1().LimitRanges(namespaceName).Create(context.Background(), limitRange, metav1.CreateOptions{})
+			if err != nil {
+				return err
+			}
+			logrus.Warn("Created LimitRange for namespace: ", namespaceName)
+		} else {
+			return err
+
+		}
+	} else {
+		// Update the existing LimitRange
+		existingLimitRange.Spec = limitRange.Spec
+		_, err = clientset.CoreV1().LimitRanges(namespaceName).Update(context.Background(), existingLimitRange, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+		logrus.Warn("Updated LimitRange for namespace: ", namespaceName)
 	}
 
-	fmt.Printf("Created LimitRange for namespace %s\n", namespaceName)
 	return nil
 }
 
+func lookupEnvOrEmpty(key string) string {
+	value, _ := os.LookupEnv(key)
+
+	return value
+}
```

Comment:
This is not needed, just use `os.Getenv` which does exactly the same thing

Reply:
I'm not sure I understand.